### PR TITLE
libxc: dont compile third/fourth derivative code

### PR DESCRIPTION
**Pull request by ehrich-pt**
A bigger software will use libxc in the future, but with this functions. The development of this software is on a macos-system.  As i see inside the ChangeLog.md

"- Added the possibility of selectively disabling the compilation of vxc, fxc, kxc, and lxc to both Autotools and CMake build systems. By default kxc and lxc are not compiled. This allows for faster compilation times."

this parameter is inside just for reducing the compiling-time.

For myself i don't use a macos system, but the changes are tested.

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/lib/libxc.rb [hunk 1]
```diff
@@ -23,6 +23,8 @@ class Libxc < Formula
   def install
     system "cmake", "-S", ".", "-B", "build",
                     "-DENABLE_FORTRAN=ON",
+                    "-DDISABLE_KXC=OFF",
+                    "-DDISABLE_LXC=OFF",
```
Comment (chenrui333) [inline Formula/lib/libxc.rb:27]: can you link some references for these changes? why is this necessary.

Comment (ehrich-pt) [reply Formula/lib/libxc.rb:27]: Hello,

it will be used inside the software 
https://github.com/cp-paw
in future and it changes the compiling-time of this library, only. 

I can imagin that other Software, like the cp-paw, also use this extensions.
In the main branch of this library, the default is to use it (if i understand the installation routine correctly ...)

Best regards,
Axel

Comment (chenrui333) [reply Formula/lib/libxc.rb:27]: Make sense, I see the changes in the devel branch as well. Thanks!

related commits:
- https://gitlab.com/libxc/libxc/-/commit/b0ce1d8eaf224c733a00f1cbd02e85d7b85bc5ff
- https://gitlab.com/libxc/libxc/-/commit/a161a63499f882b9256815c218a1f007b0a055b7

---

gonna fresh your PR to satisfy the commit style.

**Review by chenrui333 — APPROVED** [review]

**Review by timsutton — APPROVED** [review]
